File: main-workshop/services/dotnet/ride-service/Models/Ride.cs
using System.Text.Json.Serialization;



public class Ride
{
    [JsonPropertyName("rideId")]
    public string RideId { get; set; } = string.Empty;

    [JsonPropertyName("riderId")]
    public string RiderId { get; set; } = string.Empty;

    [JsonPropertyName("riderName")]
    public string RiderName { get; set; } = string.Empty;

    [JsonPropertyName("pickupLocation")]
    public Location PickupLocation { get; set; } = new();

    [JsonPropertyName("destinationLocation")]
    public Location DestinationLocation { get; set; } = new();

    // Linear status flow: requested → driver-assigned → in-progress → completed (or cancelled)
    [JsonPropertyName("status")]
    public string Status { get; set; } = "requested";

    [JsonPropertyName("driverId")]
    public string? DriverId { get; set; }

    [JsonPropertyName("driverName")]
    public string? DriverName { get; set; }

    [JsonPropertyName("finalPrice")]
    public decimal? FinalPrice { get; set; }

    [JsonPropertyName("paymentMethod")]
    public string PaymentMethod { get; set; } = "credit-card";

    [JsonPropertyName("createdAt")]
    public string CreatedAt { get; set; } = DateTime.UtcNow.ToString("O");

    [JsonPropertyName("updatedAt")]
    public string UpdatedAt { get; set; } = DateTime.UtcNow.ToString("O");

    [JsonPropertyName("deviceId")]
    public string? DeviceId { get; set; }

    override public string ToString() => RiderId;
}

public class Location
{
    [JsonPropertyName("address")]
    public string Address { get; set; } = string.Empty;

    [JsonPropertyName("latitude")]
    public double Latitude { get; set; }

    [JsonPropertyName("longitude")]
    public double Longitude { get; set; }
}

public class CreateRideRequest
{
    [JsonPropertyName("riderId")]
    public string RiderId { get; set; } = string.Empty;

    [JsonPropertyName("riderName")]
    public string RiderName { get; set; } = string.Empty;

    [JsonPropertyName("pickupLocation")]
    public Location PickupLocation { get; set; } = new();

    [JsonPropertyName("destinationLocation")]
    public Location DestinationLocation { get; set; } = new();

    [JsonPropertyName("paymentMethod")]
    public string PaymentMethod { get; set; } = "credit-card"; // credit-card, cash, somecompany-pay
}

File: main-workshop/services/java/ride-service/src/main/java/com/powertoolsride/rideservice/model/Ride.java
package com.powertoolsride.rideservice.model;

import com.fasterxml.jackson.annotation.JsonProperty;
import java.math.BigDecimal;

public record Ride(
    @JsonProperty("rideId") String rideId,
    @JsonProperty("riderId") String riderId,
    @JsonProperty("riderName") String riderName,
    @JsonProperty("pickupLocation") Location pickupLocation,
    @JsonProperty("destinationLocation") Location destinationLocation,
    @JsonProperty("status") String status,
    @JsonProperty("paymentMethod") String paymentMethod,
    @JsonProperty("deviceId") String deviceId,
    @JsonProperty("createdAt") String createdAt,
    @JsonProperty("updatedAt") String updatedAt,
    @JsonProperty("driverId") String driverId,
    @JsonProperty("driverName") String driverName,
    @JsonProperty("finalPrice") BigDecimal finalPrice
) {}
